src/package_staleness.py

```diff
@@ -0,0 +1,127 @@
+                                        x['version'].startswith(package_major_version_prefix)][-1]
+        else:
+            package_minor_version_prefix = ".".join(versioning_split[:2]) + "."
+            package_version_in_conda = [x['version'] for x in package_metadata if
+                                        x['version'].startswith(package_minor_version_prefix)][-1]
+
+        package_to_version_mapping[package] = package_version_in_conda
+    return package_to_version_mapping
+
+
+def _generate_report(package_versions_in_conda_forge, package_versions_in_image, processor,
+                     version):
+    print('\n# Staleness Report: ' + version + '(' + processor + ')\n')
+    print('Package | Current Version in the Distribution image | Latest Relevant Version in '
+          'Conda Forge')
+    print('---|---|---')
+    for package in package_versions_in_conda_forge:
+        if package_versions_in_image[package] == package_versions_in_conda_forge[package]:
+            print(package + '|' + package_versions_in_image[package] + '|' +
+                  package_versions_in_conda_forge[package])
+        else:
+            print('${\color{red}' + package + '}$' + '|' + package_versions_in_image[package] +
+                  '|' + package_versions_in_conda_forge[package])
+
+
+def _get_installed_package_versions_and_conda_versions(processor, target_version_dir,
+                                                       target_version) \
+        -> (dict[str, str], dict[str, str]):
+    env_in_file_name = processor + '.env.in'
+    env_out_file_name = processor + '.env.out'
+    required_packages_from_target = get_required_packages_from_env_in(
+        target_version_dir + "/" + env_in_file_name
+    )
+    installed_packages_from_target, package_to_conda_forge_subdirectories_mapping = \
+        get_installed_packages_and_conda_forge_sub_dir_from_env_out(
+            target_version_dir + "/" + env_out_file_name)
+    # We only care about packages which are present in env.in
+    installed_packages_from_target = {k: v for k, v in installed_packages_from_target.items()
+                                      if k in required_packages_from_target}
+    # Remove Python from the dictionary, we don't want to track python version as part of our
+    # staleness report.
+    installed_packages_from_target.pop('python', None)
+    latest_package_versions_in_conda_forge = \
+        _get_package_versions_in_conda_forge(installed_packages_from_target,
+                                             package_to_conda_forge_subdirectories_mapping,
+                                             target_version)
+    return installed_packages_from_target, latest_package_versions_in_conda_forge
+
+
+def generate_package_staleness_report(args):
+    target_version = get_semver(args.target_patch_version)
+    target_version_dir = get_dir_for_version(target_version)
+    for processor in ['cpu', 'gpu']:
```

Comment:
I think we should run a for-loop on `_image_generator_configs` instead of `['cpu', 'gpu']`. Again, trying to keep configuration (such as, what variants we support) in a single place.